src/main.cpp

```diff
@@ -497,16 +497,27 @@ int llama_generate(struct llama_context_wrapper * ctx_w, gpt_params params, py::
                 std::string line;
                 bool another_line = true;
                 do {
-                    if (!std::getline(std::cin, line)) {
-                        // input stream is bad or EOF received
+                    py::handle x = grab_text_callback();
+
+                    if (x.is_none())
+                    {
                         return 0;
                     }
-                    if (line.empty() || line.back() != '\\') {
-                        another_line = false;
-                    } else {
-                        line.pop_back(); // Remove the continue character
+                    else if (!py::isinstance<py::str>(x))
+                    {
+                        new_text_callback("Input was not of type py:str");
```

Comment:
Do we want to keep that warning message or have a different message? just before I do another commit changing the way we print it out.

Reply:
Follow the convention in this file, I would suggest:
```C
fprintf(stderr, "%s : input was not of type py::str. will ignore.\n", __func__);
```

Reply:
Good idea, just pushed the latest change.

Reply:
Great! Can you update the README file to let users know that they should return `None` from `grab_text_callback` to signal EOF? Also I trust you've already tested it.

Reply:
Updated the README with an example of how to run interactive mode as well since I saw people asking about that in the issues and a quick start will help. And yes, I have been running this on my end and recompiling and testing with the code snippet above each time.

Reply:
Thanks @Kasimir123 and @nuance1979  for the contribution & sorry for this late reply. 
I will merge it for now.